# Empire Remote WMIC Add User

## Description: This dataset represents an adversary remotely executing code via WMI to ad a backdoor user on the target system. This dataset focuses on the use of the WMI Win32_Process class and method Create to execute code remotely.

https://securitydatasets.com/notebooks/atomic/windows/execution/SDWIN-200914080546.html



In [112]:
procs = GET process from https://raw.githubusercontent.com/OTRF/Security-Datasets/master/datasets/atomic/windows/lateral_movement/host/empire_wmic_add_user_backdoor.zip
where command_line LIKE '%'

Block Executed in 21 seconds 
 
 
 VARIABLE 
 TYPE 
 #(ENTITIES) 
 #(RECORDS) 
 artifact* 
 directory* 
 file* 
 ipv4-addr* 
 network-traffic* 
 process* 
 user-account* 
 windows-registry-key* 
 x-oca-asset* 
 x-oca-event* 
 
 
 
 
 procs 
 process 
 30 
 7937 
 7859 
 8309 
 10571 
 48 
 24 
 7957 
 136 
 4956 
 7859 
 7859 
 
 
 *Number of related records cached.

## Find processes that could create a backdoor and end with "exe"

In [113]:
filtered_procs = GET process FROM procs WHERE [process:command_line LIKE '%exe%']

Block Executed in 1 seconds 
 
 
 VARIABLE 
 TYPE 
 #(ENTITIES) 
 #(RECORDS) 
 artifact* 
 directory* 
 file* 
 ipv4-addr* 
 network-traffic* 
 process* 
 user-account* 
 windows-registry-key* 
 x-oca-asset* 
 x-oca-event* 
 
 
 
 
 filtered_procs 
 process 
 70 
 9136 
 0 
 0 
 0 
 0 
 0 
 0 
 0 
 0 
 0 
 0 
 
 
 *Number of related records cached.

## Get an overview of all processes with "exe" as extension

In [114]:
DISP filtered_procs

pid 
 name 
 id 
 command_line 
 x_unique_id 
 created 
 binary_ref.name 
 binary_ref.id 
 binary_ref.hashes.'SHA-1' 
 binary_ref.hashes.MD5 
 binary_ref.hashes.'SHA-256' 
 binary_ref.parent_directory_ref.path 
 binary_ref.parent_directory_ref.id 
 parent_ref.pid 
 parent_ref.name 
 parent_ref.id 
 parent_ref.command_line 
 parent_ref.x_unique_id 
 parent_ref.created 
 parent_ref.parent_ref.pid 
 parent_ref.parent_ref.name 
 parent_ref.parent_ref.id 
 parent_ref.parent_ref.command_line 
 parent_ref.parent_ref.x_unique_id 
 parent_ref.parent_ref.created 
 parent_ref.binary_ref.name 
 parent_ref.binary_ref.id 
 parent_ref.binary_ref.hashes.'SHA-1' 
 parent_ref.binary_ref.hashes.MD5 
 parent_ref.binary_ref.hashes.'SHA-256' 
 parent_ref.binary_ref.parent_directory_ref.path 
 parent_ref.binary_ref.parent_directory_ref.id 
 parent_ref.creator_user_ref.user_id 
 parent_ref.creator_user_ref.id 
 creator_user_ref.user_id 
 creator_user_ref.id 
 
 
 
 
 11124 
 netsh.exe 
 process--56ce8501-e1c0-55e5-bafe-7094e01e2823 
 "C:\windows\system32\netsh.exe" advfirewall firewall add rule name=python.exe dir=in action=allow description=python.exe program=C:\users\wardog\appdata\local\programs\python\python39\python.exe enable=yes localport=any protocol=tcp remoteip=any 
 None 
 None 
 netsh.exe 
 file--b79adcff-5054-5b1f-96af-500475b71f21 
 None 
 None 
 None 
 C:\Windows\System32 
 directory--0a58d0c1-59e6-5afd-8252-dcd3f13e5622 
 9320.0 
 powershell.exe 
 process--31c1efcf-ee2a-55ec-948a-b900f7b6eac1 
 None 
 None 
 None 
 
 None 
 None 
 None 
 None 
 None 
 powershell.exe 
 file--0daceb39-1883-534b-b6ab-ebcafab74cfa 
 None 
 None 
 None 
 C:\Windows\System32\WindowsPowerShell\v1.0 
 directory--c18706ba-12ab-53b2-9524-6ef585f3127a 
 None 
 None 
 wardog 
 user-account--7c7b7450-5cfd-5683-a891-20644b37e384 
 
 
 4756 
 netsh.exe 
 process--a30e10bc-9343-50f6-881f-f5dd2cfb8404 
 "C:\windows\system32\netsh.exe" advfirewall firewall add rule name=python.exe dir=in action=allow description=python.exe program=C:\users\wardog\appdata\local\programs\python\python39\python.exe enable=yes localport=any protocol=udp remoteip=any 
 None 
 None 
 netsh.exe 
 file--b79adcff-5054-5b1f-96af-500475b71f21 
 None 
 None 
 None 
 C:\Windows\System32 
 directory--0a58d0c1-59e6-5afd-8252-dcd3f13e5622 
 9320.0 
 powershell.exe 
 process--cfd78b3d-710f-5bee-a2ee-64d2184e5e2a 
 None 
 None 
 None 
 
 None 
 None 
 None 
 None 
 None 
 powershell.exe 
 file--0daceb39-1883-534b-b6ab-ebcafab74cfa 
 None 
 None 
 None 
 C:\Windows\System32\WindowsPowerShell\v1.0 
 directory--c18706ba-12ab-53b2-9524-6ef585f3127a 
 None 
 None 
 wardog 
 user-account--7c7b7450-5cfd-5683-a891-20644b37e384 
 
 
 10220 
 python.exe 
 process--cbb17e84-ebed-5c83-bafa-bc89d5e765aa 
 "C:\Users\wardog\AppData\Local\Programs\Python\Python39\python.exe" -m http.server 8000 
 None 
 None 
 python.exe 
 file--4a09f4fb-8fac-5ea5-983d-eadc939036a7 
 None 
 None 
 None 
 C:\Users\wardog\AppData\Local\Programs\Python\Python39 
 directory--dcd4ce59-3619-5b4d-8f85-856d8cdc32ef 
 9320.0 
 powershell.exe 
 process--de1a58b9-64c6-58dc-8383-f9061f1e2f0b 
 None 
 None 
 None 
 
 None 
 None 
 None 
 None 
 None 
 powershell.exe 
 file--0daceb39-1883-534b-b6ab-ebcafab74cfa 
 None 
 None 
 None 
 C:\Windows\System32\WindowsPowerShell\v1.0 
 directory--c18706ba-12ab-53b2-9524-6ef585f3127a 
 None 
 None 
 wardog 
 user-account--7c7b7450-5cfd-5683-a891-20644b37e384 
 
 
 11124 
 netsh.exe 
 process--dd2e0895-a377-5fbc-9c7c-6222351f01ff 
 "C:\windows\system32\netsh.exe" advfirewall firewall add rule name=python.exe dir=in action=allow description=python.exe program=C:\users\wardog\appdata\local\programs\python\python39\python.exe enable=yes localport=any protocol=tcp remoteip=any 
 {39e4a257-4209-5f9a-3d36-000000000700} 
 2020-10-29T04:16:09.914Z 
 netsh.exe 
 file--b79adcff-5054-5b1f-96af-500475b71f21 
 None 
 None 
 None 
 C:\Windows\System32 
 directory--0a58d0c1-59e6-5afd-8252-dcd3f13e5622 
 9320.0 
 powershell.

## See which parent process has executed a command with "backdoor"

In [115]:
backdoor = GET process FROM filtered_procs WHERE parent_ref.command_line LIKE '%backdoor%'

Block Executed in 1 seconds 
 
 
 VARIABLE 
 TYPE 
 #(ENTITIES) 
 #(RECORDS) 
 artifact* 
 directory* 
 file* 
 ipv4-addr* 
 network-traffic* 
 process* 
 user-account* 
 windows-registry-key* 
 x-oca-asset* 
 x-oca-event* 
 
 
 
 
 backdoor 
 process 
 2 
 120 
 0 
 0 
 0 
 0 
 0 
 0 
 0 
 0 
 0 
 0 
 
 
 *Number of related records cached.

## Get a better overview of the two entities

In [116]:
DISP backdoor
INFO backdoor

pid 
 name 
 id 
 command_line 
 x_unique_id 
 created 
 binary_ref.name 
 binary_ref.id 
 binary_ref.hashes.'SHA-1' 
 binary_ref.hashes.MD5 
 binary_ref.hashes.'SHA-256' 
 binary_ref.parent_directory_ref.path 
 binary_ref.parent_directory_ref.id 
 parent_ref.pid 
 parent_ref.name 
 parent_ref.id 
 parent_ref.command_line 
 parent_ref.x_unique_id 
 parent_ref.created 
 parent_ref.parent_ref.pid 
 parent_ref.parent_ref.name 
 parent_ref.parent_ref.id 
 parent_ref.parent_ref.command_line 
 parent_ref.parent_ref.x_unique_id 
 parent_ref.parent_ref.created 
 parent_ref.binary_ref.name 
 parent_ref.binary_ref.id 
 parent_ref.binary_ref.hashes.'SHA-1' 
 parent_ref.binary_ref.hashes.MD5 
 parent_ref.binary_ref.hashes.'SHA-256' 
 parent_ref.binary_ref.parent_directory_ref.path 
 parent_ref.binary_ref.parent_directory_ref.id 
 parent_ref.creator_user_ref.user_id 
 parent_ref.creator_user_ref.id 
 creator_user_ref.user_id 
 creator_user_ref.id 
 
 
 
 
 7264 
 conhost.exe 
 process--97809f86-d9d8-5236-8969-fa4d0293771c 
 \??\C:\windows\system32\conhost.exe 0xffffffff -ForceV1 
 {83d0c8c3-5caa-5f5f-f202-000000000400} 
 2020-09-14T12:06:02.332Z 
 conhost.exe 
 file--9d6a9721-9948-53ff-83f8-3865bc03c727 
 None 
 None 
 None 
 C:\Windows\System32 
 directory--0a58d0c1-59e6-5afd-8252-dcd3f13e5622 
 7768 
 net.exe 
 process--b421d3ba-0de0-54b2-958d-465fd5d873c1 
 net user /add backdoor paw0rd1 
 {83d0c8c3-5caa-5f5f-f102-000000000400} 
 2020-09-14T12:06:02.316Z 
 3952 
 WmiPrvSE.exe 
 process--9c817e8e-144f-5ef9-8d59-8eb97ae5853f 
 C:\windows\system32\wbem\wmiprvse.exe -secured -Embedding 
 {83d0c8c3-5caa-5f5f-f002-000000000400} 
 2020-09-14T12:06:02.209Z 
 net.exe 
 file--50a5c5cc-7636-5385-96d3-7d14068a4c10 
 None 
 None 
 None 
 C:\Windows\System32 
 directory--0a58d0c1-59e6-5afd-8252-dcd3f13e5622 
 THESHIRE\pgustavo 
 user-account--2dfb4c1a-ebd5-5c42-a8f1-26acde5c50eb 
 THESHIRE\pgustavo 
 user-account--2dfb4c1a-ebd5-5c42-a8f1-26acde5c50eb 
 
 
 3692 
 net1.exe 
 process--b635c63a-63fd-5edb-bfad-67fd995634a6 
 C:\windows\system32\net1 user /add backdoor paw0rd1 
 {83d0c8c3-5caa-5f5f-f302-000000000400} 
 2020-09-14T12:06:02.404Z 
 net1.exe 
 file--fc872882-afc7-59af-b296-36774204cd2b 
 None 
 None 
 None 
 C:\Windows\System32 
 directory--0a58d0c1-59e6-5afd-8252-dcd3f13e5622 
 7768 
 net.exe 
 process--b421d3ba-0de0-54b2-958d-465fd5d873c1 
 net user /add backdoor paw0rd1 
 {83d0c8c3-5caa-5f5f-f102-000000000400} 
 2020-09-14T12:06:02.316Z 
 3952 
 WmiPrvSE.exe 
 process--9c817e8e-144f-5ef9-8d59-8eb97ae5853f 
 C:\windows\system32\wbem\wmiprvse.exe -secured -Embedding 
 {83d0c8c3-5caa-5f5f-f002-000000000400} 
 2020-09-14T12:06:02.209Z 
 net.exe 
 file--50a5c5cc-7636-5385-96d3-7d14068a4c10 
 None 
 None 
 None 
 C:\Windows\System32 
 directory--0a58d0c1-59e6-5afd-8252-dcd3f13e5622 
 THESHIRE\pgustavo 
 user-account--2dfb4c1a-ebd5-5c42-a8f1-26acde5c50eb 
 THESHIRE\pgustavo 
 user-account--2dfb4c1a-ebd5-5c42-a8f1-26acde5c50eb 
 
 
 
 
 
 
 Entity Type 
 process 
 
 
 Number of Entities 
 2 
 
 
 Number of Records 
 120 
 
 
 Entity Attributes 
 pid, name, id, command_line, created 
 
 
 Indirect Attributes 
 [] 
 
 
 Customized Attributes 
 x_unique_id 
 
 
 Birth Command 
 get 
 
 
 Associated Datasource 
 https://raw.githubusercontent.com/OTRF/Security-Datasets/master/datasets/atomic/windows/lateral_movement/host/empire_wmic_add_user_backdoor.zip 
 
 
 Dependent Variables 
 filtered_procs

## It does not appear to be the case that network traffic was generated in relation to the backdoor process. 

In [117]:
nt = FIND network-traffic CREATED BY backdoor

Block Executed in 1 seconds 
 
 
 VARIABLE 
 TYPE 
 #(ENTITIES) 
 #(RECORDS) 
 artifact* 
 directory* 
 file* 
 ipv4-addr* 
 network-traffic* 
 process* 
 user-account* 
 windows-registry-key* 
 x-oca-asset* 
 x-oca-event* 
 
 
 
 
 nt 
 network-traffic 
 0 
 0 
 0 
 0 
 0 
 0 
 0 
 0 
 0 
 0 
 0 
 0 
 
 
 *Number of related records cached.

In [118]:
parent_procs = FIND process CREATED backdoor

Block Executed in 1 seconds 
 
 
 VARIABLE 
 TYPE 
 #(ENTITIES) 
 #(RECORDS) 
 artifact* 
 directory* 
 file* 
 ipv4-addr* 
 network-traffic* 
 process* 
 user-account* 
 windows-registry-key* 
 x-oca-asset* 
 x-oca-event* 
 
 
 
 
 parent_procs 
 process 
 1 
 87 
 7859 
 8309 
 10571 
 48 
 24 
 7986 
 136 
 4956 
 7859 
 7859 
 
 
 *Number of related records cached.

## Process 3692 shows that a user named backdoor has been added

In [119]:
DISP backdoor ATTR pid, name, id, command_line, created

pid 
 name 
 id 
 command_line 
 created 
 
 
 
 
 7264 
 conhost.exe 
 process--97809f86-d9d8-5236-8969-fa4d0293771c 
 \??\C:\windows\system32\conhost.exe 0xffffffff -ForceV1 
 2020-09-14T12:06:02.332Z 
 
 
 3692 
 net1.exe 
 process--b635c63a-63fd-5edb-bfad-67fd995634a6 
 C:\windows\system32\net1 user /add backdoor paw0rd1 
 2020-09-14T12:06:02.404Z